pkg/schedule/schedulers/balance_range.go

```diff
@@ -617,88 +624,65 @@ func (s *balanceRangeScheduler) prepare(cluster sche.SchedulerCluster, opInfluen
-		region:           nil,
 		averageScore:     averageScore,
 		job:              job,
-		opInfluence:      opInfluence,
-		tolerate:         tolerantSizeRatio,
+		solver:           solver,
 	}, nil
 }
 
-func (p *balanceRangeSchedulerPlan) sourceStoreID() uint64 {
-	return p.source.GetID()
-}
-
-func (p *balanceRangeSchedulerPlan) targetStoreID() uint64 {
-	return p.target.GetID()
-}
-
-func (p *balanceRangeSchedulerPlan) score(storeID uint64) int64 {
+func (p *balanceRangeSchedulerPlan) score(storeID uint64) float64 {
 	return p.scoreMap[storeID]
 }
 
-func (p *balanceRangeSchedulerPlan) shouldBalance(scheduler string) bool {
-	sourceInfluence := p.opInfluence.GetStoreInfluence(p.sourceStoreID())
-	sourceInf := sourceInfluence.GetStoreInfluenceByRole(p.job.Rule)
-	// Sometimes, there are many remove-peer operators in the source store, we don't want to pick this store as source.
-	if sourceInf < 0 {
-		sourceInf = -sourceInf
+func (p *balanceRangeSchedulerPlan) sourceScore(storeID uint64) float64 {
+	originScore := p.score(storeID)
+	s := p.solver
+	tolerantResource := s.getTolerantResource()
+	// to avoid schedule too much, if A's core greater than B and C a little
+	// we want that A should be moved out one region not two
+	influence := s.getOpInfluence(storeID)
+	if influence > 0 {
+		influence = -influence
 	}
+	return originScore + float64(influence-tolerantResource)
+}
+
+func (p *balanceRangeSchedulerPlan) targetScore(storeID uint64) float64 {
+	originScore := p.score(storeID)
+	s := p.solver
+	tolerantResource := s.getTolerantResource()
 	// to avoid schedule too much, if A's core greater than B and C a little
 	// we want that A should be moved out one region not two
-	sourceScore := p.sourceScore - sourceInf - p.tolerate
-
-	targetInfluence := p.opInfluence.GetStoreInfluence(p.targetStoreID())
-	targetInf := targetInfluence.GetStoreInfluenceByRole(p.job.Rule)
-	// Sometimes, there are many add-peer operators in the target store, we don't want to pick this store as target.
-	if targetInf < 0 {
-		targetInf = -targetInf
-	}
-	targetScore := p.targetScore + targetInf + p.tolerate
-
-	// the source score must be greater than the target score
-	shouldBalance := sourceScore >= targetScore
-	if !shouldBalance && log.GetLevel() <= zap.DebugLevel {
-		log.Debug("skip balance",
-			zap.String("scheduler", scheduler),
-			zap.Uint64("region-id", p.region.GetID()),
-			zap.Uint64("source-store", p.sourceStoreID()),
-			zap.Uint64("target-store", p.targetStoreID()),
-			zap.Int64("origin-source-score", p.sourceScore),
-			zap.Int64("origin-target-score", p.targetScore),
-			zap.Int64("influence-source-score", sourceScore),
-			zap.Int64("influence-target-score", targetScore),
-			zap.Int64("average-region-score", p.averageScore),
-			zap.Int64("tolerate", p.tolerate),
-		)
+	influence := s.getOpInfluence(storeID)
+	if influence < 0 {
+		influence = -influence
 	}
-	return shouldBalance
+	return originScore + float64(influence-tolerantResource)
```

Comment:
In previous, we use `targetScore := p.targetScore + targetInf + p.tolerate`, but now we reduce `tolerantResource`

Reply:
yes, good catch